### DISCUSSION TITLE
国内访问官方源地址不方便，是否可以增加换源的脚本，让用户在编译期就自定义

### DISCUSSION THREAD

**Discussion by wanmyj** [chenmozhijin/OpenWrt-K#46] [Ideas]
RT
官方源是 `https://downloads.openwrt.org`. 
国内源有很多，`https://dl.openwrt.ai/`比较流行。
虽然可以写入包以后再自定义，或者用户配置备份恢复，但是如果能在编译时候直接指定，会更方便一些

**Comment by chenmozhijin** [comment]
你可以直接在[zzz-chenmozhijin](https://github.com/chenmozhijin/OpenWrt-K/blob/main/files/etc/uci-defaults/zzz-chenmozhijin)中添加你的换源脚本，这个脚本会在第一次开机时或系统升级/重置后运行。

当然在[uci-defaults](https://github.com/chenmozhijin/OpenWrt-K/tree/main/files/etc/uci-defaults)文件夹下直接添加换源脚本也可以，不过需要注意文件名，因为系统是按文件名的顺序决定脚本运行顺序的（建议以“zzz-”开头）。


[openwrt官方的uci-defaults文档](https://openwrt.org/docs/guide-developer/uci-defaults)

[定制编译 OpenWrt 固件](https://github.com/chenmozhijin/OpenWrt-K/wiki/%E5%AE%9A%E5%88%B6%E7%BC%96%E8%AF%91-OpenWrt-%E5%9B%BA%E4%BB%B6#5%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9)注意事项5

**Reply by wanmyj** [reply]
感谢！还是得仔细看看文档
